### PR TITLE
Fix invalid delivery mode returned by AMQPEnvelop::getDeliveryMode

### PR DESCRIPTION
**Pull request by pinepain**
This PR:

 - Fix issue #170: invalid delivery mode `0` returned by `AMQPEnvelop::getDeliveryMode()` when no `delivery_mode` message property set (that means that delivery mode is 1 - transient).
 - Add support for [new Travis container-based infrastructure](http://docs.travis-ci.com/user/migrating-from-legacy/).
 - Add [rabbitmq-c v0.7.0](https://github.com/alanxz/rabbitmq-c/releases/tag/v0.7.0) to build matrix.

This changes are BC compatible (though it is highly recommended to test integration before update to newer php-amqp version that will include this changes, as always, actually).

### CODE REVIEW

**Comment by lstrojny** [comment]
Can you check the build?

**Comment by pinepain** [comment]
Yeah, working on it, something strange goes on - in my branch it is green while PR build red.

**Comment by pinepain** [comment]
Done

**Comment by pinepain** [comment]
@lstrojny What I did is interpreted missed `delivery_mode` as transient (1) which is mostly correct for RabbitMQ but incorrect for ActiveMQ. While this should work perfect for RabbitMQ and doesn't break BC, compatibility with other MQ brokers may will be lost.

I also experimented with ignoring `delivery_mode` property when it is `NULL`, which may somehow break BC (`AMQPEnvelop::getDeliveryMode()` will also return `null`, when delivery mode is unknown) - same like in Java API - [AMQP.BasicProperties.getDeliveryMode()](http://www.rabbitmq.com/releases/rabbitmq-java-client/v3.5.4/rabbitmq-java-client-javadoc-3.5.4/com/rabbitmq/client/AMQP.BasicProperties.html#getDeliveryMode%28%29).

So until we figure out which way is the most appropriate, I mark this PR as WIP.

**Comment by stof** [comment]
@pinepain in my application, all messages are delivered as persistent (2) and I was still missing the delivery mode. So the interpretation of missing delivery mode looks weird.

**Comment by stof** [comment]
ah, these messages were not persistent. Sorry for the wrong report.

**Comment by pinepain** [comment]
The way with returning `NULL` for values that was not originally set looks more logical for me, so I would even propose to do that for all getters, not only for delivery mode. While it will be less confusing, it will be also consistent with Java API, which looks for me well-designed. The only part I really worry is BC.

**Comment by stof** [comment]
it would be great if RabbitMQ was sending the delivery_mode flag even when it matches the default, as it seems that the default value is not part of the AMQP protocol...

@pinepain currently, it returns an invalid value in such case, which breaks the dispatching of a message reusing the delivery mode, and does not match any constant. So it looks fine to me (please make sure that passing ``null`` as delivery mode when submitting is supported, letting the server choose the default delivery mode, to be consistent)

**Comment by pinepain** [comment]
Returning `null` from `AMQPEnvelop::getDeliveryMode` will definitely break BC - strict checks will fail. 

Looking through ActiveMQ docs and specs gives me that it support only AMQP 1.0 so from 0.9.1 we have RabbitMQ, Apache Qpid and OpenAMQ (doesn't look active at all). While I mostly familiar with RabbitMQ, brief googling gives me that Qpid operates delivery mode on client side and looks like that OpenAMQ too. So my suggestion about breaking BC ruined and I guess we can stick with suggesting default `delivery_mode` as transient OR even setting `delivery_mode` as transient on sending if it is not set (but it looks ambiguous for me and we may work in heterogeneous environment where other clients in other languages doesn't send any delivery_mode explicitly).

**Comment by pinepain** [comment]
The only functional change that affects issue #170 is https://github.com/pinepain/php-amqp/commit/614cab5b5e4c50a7ba836aceb22593572a806ecb#diff-9b81a369f4ea42ead436323d99882d90R298: 

```diff
@@ -295,6 +295,8 @@ void convert_amqp_envelope_to_zval(amqp_envelope_t *amqp_envelope, zval *envelop
 
 	if (p->_flags & AMQP_BASIC_DELIVERY_MODE_FLAG) {
 		AMQP_SET_LONG_PROPERTY(envelope->delivery_mode, p->delivery_mode);
+	} else {
+		AMQP_SET_LONG_PROPERTY(envelope->delivery_mode, AMQP_DELIVERY_NONPERSISTENT);
 	}
 
 	if (p->_flags & AMQP_BASIC_PRIORITY_FLAG) { 
```

**Comment by pinepain** [comment]
@lstrojny I guess we can safely merge this changes

File: .travis.yml [hunk 1]
```diff
@@ -17,15 +19,15 @@ matrix:
...
-  - sudo service rabbitmq-server restart
+#install:
+#  - sudo apt-get update
+#  - sudo apt-get -qq --fix-missing install valgrind
+#
+#  - wget -qO - http://www.rabbitmq.com/rabbitmq-signing-key-public.asc | sudo apt-key add -
+#  - sudo add-apt-repository 'deb http://www.rabbitmq.com/debian/ testing main'
+#  - sudo apt-get update
+#  - sudo apt-get install --only-upgrade -y rabbitmq-server
+#  - sudo service rabbitmq-server restart
```
Comment (stof) [inline .travis.yml]: this should be removed rather than being commented. The history is already kept by git, and it is much better at it than comments

Comment (pinepain) [inline .travis.yml]: True. The original intention is to keep it here to be able to switch to newer RabbitMQ server version during development (while Travis machines install RabbitMQ from the same source, it is often outdated), so I guess it should be kept for now.

**Comment by pinepain** [comment]
@lstrojny can we merge this and release?

**Comment by lstrojny** [comment]
For sure! @pinepain do you want to ship another beta?

**Comment by pinepain** [comment]
Thanks! @lstrojny let's have extra one beta just in case we missed something. Stable should be stable.

**Comment by lstrojny** [comment]
OK!

**Comment by pinepain** [comment]
@lstrojny JFYI: you tagged beta4 but didn't released on PECL.

**Comment by lstrojny** [comment]
@pinepain I know, I am having trouble with the PECL account. Can login to other php.net services but not there. Trying to get help.

**Pull request merged**